tools/github-actions/audit/src/main.ts

```diff
@@ -1,137 +1,55 @@
-  core.info('Computing Report for Yarn 3');
-  const reportJson = JSON.parse(response) as Audit;
-  core.debug(response);
-  const nbVulnerabilities = Object.values(reportJson.metadata.vulnerabilities as { [key: string]: number } || {}).reduce((acc, curr) => acc + curr, 0);
-  let highestSeverityFound: Severity | undefined;
-  for (let index = severities.length; index >= 0; index--) {
-    const severity: Severity = severities[index];
-    if (reportJson.metadata.vulnerabilities[severity] > 0) {
-      highestSeverityFound = severity;
-      break;
-    }
-  }
-  return Object.values(reportJson.advisories as Readonly<Record<GitHubAdvisoryId, Advisory>>)
-    .reduce<OtterAuditReport>((currentVulnerabilities, advisory: Advisory) => {
-      core.info(`${severities.indexOf(severityThreshold)} - ${severities.indexOf(advisory.severity)}`);
-      if (severities.indexOf(severityThreshold) <= severities.indexOf(advisory.severity)) {
-        currentVulnerabilities.errors.push({severity: advisory.severity, overview: advisory.overview, moduleName: advisory.module_name});
-      } else {
-        currentVulnerabilities.warnings.push({severity: advisory.severity, overview: advisory.overview, moduleName: advisory.module_name});
-      }
-      return currentVulnerabilities;
-    }, {errors: [], warnings: [], nbVulnerabilities, highestSeverityFound} as OtterAuditReport);
-}
+const colors = ['', 'green', 'yellow', 'orange', 'red'];
 
 async function run(): Promise<void> {
 
   try {
+    const cwd = process.env.GITHUB_WORKSPACE!;
+    const packageManager = fs.existsSync(path.resolve(cwd, 'yarn.lock')) ? 'yarn' : 'npm';
     const severityConfig = core.getInput('severity') as Severity;
     const allWorkspaces = core.getInput('allWorkspaces') === 'true';
     const recursive = core.getInput('recursive') === 'true';
     const environment = core.getInput('environment');
-    const versionOutput = await getExecOutput('yarn --version', [], {cwd: process.env.GITHUB_WORKSPACE});
-    const version = Number.parseInt(versionOutput.stdout.split('.')[0], 10);
-    const command = `yarn npm audit --environment ${environment} ${allWorkspaces ? '--all ' : ''}${recursive ? '--recursive ' : ''}--json`;
+    let auditReporter: OtterAuditReporter;
+    let auditCommand: string;
+    if (packageManager === 'yarn') {
+      const versionOutput = await getExecOutput('yarn --version', [], {cwd: process.env.GITHUB_WORKSPACE});
+      const version = Number.parseInt(versionOutput.stdout.split('.')[0], 10);
+      auditCommand = version <= 1 ?
+        `yarn audit ${environment === 'production' ? '--groups "dependencies peerDependencies" ' : ''}--json` :
+        `yarn npm audit --environment ${environment} ${allWorkspaces ? '--all ' : ''}${recursive ? '--recursive ' : ''}--json`;
+      auditReporter = version >= 4 ? computeYarn4Report : version >= 2 ? computeYarn3Report : computeYarn1Report;
```

Comment:
would it be more readable in a switch case?

Reply:
I find this version easily readable